async-nats/src/service/endpoint.rs

```diff
@@ -177,13 +170,59 @@ pub struct Stats {
     #[serde(default, with = "serde_nanos")]
     pub average_processing_time: std::time::Duration,
     /// Last error that occurred.
+    #[serde(with = "serde_error_string")]
     pub last_error: Option<error::Error>,
     /// Custom data added by [crate::service::Config::stats_handler]
-    pub data: String,
+    pub data: Option<serde_json::Value>,
     /// Queue group to which this endpoint is assigned to.
     pub queue_group: String,
 }
 
+mod serde_error_string {
```

Comment:
As we have our own error type, the serde impl can be attached to it directly.

Reply:
as discussed, as a first pass, this is ok, as we would need some custom logic for serializing into "" instead of null in case of None etc.